### PR TITLE
remove id & use random number as key

### PR DESCRIPTION
**Pull request by aryani1**

### CODE REVIEW

**Comment by testower** [comment]
I don't understand the reason / motivation for this change.

**Comment by aryani1** [comment]
> I don't understand the reason / motivation for this change.

Using the index as `key` yields some unintended behaviour when deleting a `DayTypeAssignment`, e.g it looks like one is deleting the wrong assignment since deleting an item changes the indices of succeeding items. We tried therefore to generate a unique id for each item and use it as key, but it got unnecessarily complicated. 

So for now the quickest solution seemed to be to just re-render all items when something changes.

File: src/scenes/Lines/scenes/Editor/JourneyPatterns/Editor/ServiceJourneys/Editor/DayTypeEditor/DayTypeAssignments/index.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ class DayTypeAssignmentsEditor extends Component {
         {dayTypeAssignments.length > 0 ? (
           dayTypeAssignments.map((dta, i) => (
             <DayTypeAssignmentEditor
-              key={dta.id}
+              key={Math.random()}
```
Comment (FredrikMeyer) [inline src/scenes/Lines/scenes/Editor/JourneyPatterns/Editor/ServiceJourneys/Editor/DayTypeEditor/DayTypeAssignments/index.js]: Hva med bare JSON.stringify? Da er det ihvertfall deterministisk?

**Comment by testower** [comment]
> > I don't understand the reason / motivation for this change.
> 
> Using the index as `key` yields some unintended behaviour when deleting a `DayTypeAssignment`, e.g it looks like one is deleting the wrong assignment since deleting an item changes the indices of succeeding items. We tried therefore to generate a unique id for each item and use it as key, but it got unnecessarily complicated.
> 
> So for now the quickest solution seemed to be to just re-render all items when something changes.

I see, just be aware that it could cause performance degradation and lost state in child components, which could lead to bugs.

**Comment by christianmosveen** [comment]
What do you think about propagating the actual identifiers for DayTypeAssignments (and other similar objects that currently are fetched without any identifier) from the backend up to the editor, @testower?

**Comment by testower** [comment]
> What do you think about propagating the actual identifiers for DayTypeAssignments (and other similar objects that currently are fetched without any identifier) from the backend up to the editor, @testower?

Hmm yes, it's probably possible, but it seems intentionally left out, so it should be thoroughly tested to make sure it doesn't lead to bugs. I don't have enough experience with the code base to guarantee it won't cause issues :-)

You can test it locally if you want by changing this line https://github.com/entur/uttu/blob/master/src/main/java/no/entur/uttu/graphql/FlexibleLinesGraphQLSchema.java#L299

to 

```
dayTypeAssignmentObjectType = newObject(identifiedEntityObjectType).name("DayTypeAssignment")
```

**Comment by aryani1** [comment]
I'll merge this in for now as a temporary fix, and then later on try out the proposed change! Thanks for the feedback.

**Pull request merged**